File: mahaam-api-java/src/main/java/mahaam/infra/Email.java
package mahaam.infra;

import com.twilio.Twilio;
import com.twilio.rest.verify.v2.service.Verification;
import com.twilio.rest.verify.v2.service.VerificationCheck;

public class Email {
	private static String verificationServiceSid;

	public static void init(String accountSid, String verificationServiceSid, String authToken) {
		try {
			Email.verificationServiceSid = verificationServiceSid;
			Twilio.init(accountSid, authToken);
		} catch (Exception e) {
			Log.error(e.toString());
		}
	}

	public static String sendOtp(String email) {
		try {
			Verification verification = Verification.creator(verificationServiceSid, email, "email").create();
			return verification.getSid();
		} catch (Exception e) {
			Log.error(e.toString());
		}
		return null;
	}

	public static String verifyOtp(String otp, String sid, String email) {
		try {
			VerificationCheck check = VerificationCheck.creator(verificationServiceSid).setTo(email).setCode(otp)
					.create();
			return check.getStatus().toString();
		} catch (Exception e) {
			Log.error(e.toString());
		}
		return null;
	}
}

File: mahaam-api-cs/Src/Feat/User/Email.cs

using Mahaam.Infra;
using Twilio;
using Twilio.Rest.Verify.V2.Service;

namespace Mahaam.Feat.Users;

public interface IEmail
{
	void Init();
	string? SendOtp(string email);
	string? VerifyOtp(string otp, string sid, string email);
}
public class Email(ILog log, Settings settings) : IEmail
{

	public void Init()
	{
		try
		{
			TwilioClient.Init(settings.Email.AccountSid, settings.Email.AuthToken);
		}
		catch (Exception e)
		{
			log.Error(e.ToString());
		}
	}

	public string? SendOtp(string email)
	{
		try
		{
			var verification = VerificationResource.Create(pathServiceSid: settings.Email.VerificationServiceSid, to: email, channel: "email");
			return verification.Sid;
		}
		catch (Exception e)
		{
			log.Error(e.ToString());
		}
		return null;
	}

	public string VerifyOtp(string otp, string sid, string email)
	{
		try
		{
			var check = VerificationCheckResource.Create(to: email, code: otp, verificationSid: sid, pathServiceSid: settings.Email.VerificationServiceSid
			// ,verificationSid:sid was not there
			);
			return check.Status;
		}
		catch (Exception e)
		{
			log.Error(e.ToString());
		}
		return null;
	}
}